flake.nix

```diff
@@ -0,0 +1,92 @@
+{
+  # Use 'nix flake show' to discover the structure of the output.
+  # Multiple versions of compiler is supported.
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixpkgs-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = inputs:
+    let
+      cabalPackages = [ "timeline" "timeline-hedgehog" "timeline-tests" ];
+      supportedCompilers = [ "ghc8107" "ghc926" "ghc944" ];
+      defaultCompiler = "ghc926";
```

Comment:
@JackKelly-Bellroy @lrworth I'm satisfied with this configuration. Maybe we can share with our other public repositories.
<img width="582" alt="image" src="https://user-images.githubusercontent.com/16440269/223625647-11d1df70-a2e9-43a6-832e-50404590884e.png">

Reply:
Looks like an improvement on https://github.com/bellroy/haskell-cached-io/pull/10/files#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0 :) I was thinking that pattern could be put in some common bellroy flake and imported everywhere.

Reply:
Once you have something you like, you could riff on something like https://git.sr.ht/~jack/simple-haskell-flake and generalise it to multiple packages. Then you can import it into each FOSS package by flake reference instead of copying boilerplate.